### PR TITLE
all: allow renaming team and updating all references to the old name

### PR DESCRIPTION
**Pull request by cezarsa**

### CODE REVIEW

File: api/auth.go [hunk 1]
```diff
@@ -226,6 +232,108 @@ func resetPassword(w http.ResponseWriter, r *http.Request) (err error) {
...
+	})
+	if err != nil {
+		return err
+	}
+	defer func() { evt.Done(err) }()
+	u, err := t.User()
+	if err != nil {
+		return err
+	}
+	defer func() {
```
Comment (ggarnier) [inline api/auth.go]: Shouldn't this `defer` be defined after `CreateTeam`?

Comment (cezarsa) [inline api/auth.go]: Right on, fixed now.

File: api/auth.go [hunk 2]
```diff
@@ -226,6 +232,99 @@ func resetPassword(w http.ResponseWriter, r *http.Request) (err error) {
...
+	volume.RenameTeam,
+	provision.RenamePoolTeam,
+}
+
+// title: team update
+// path: /teams/{name}
+// method: POST
+// consume: application/x-www-form-urlencoded
+// responses:
+//   200: Team created
```
Comment (ArxdSilva) [inline api/auth.go]: shouldnt this be 'team updated'?

Comment (cezarsa) [inline api/auth.go]: fixed!

**Pull request merged**